src/libstore/unix/build/local-derivation-goal.cc

```diff
@@ -1504,7 +1505,7 @@ void LocalDerivationGoal::startDaemon()
 
     daemonThread = std::thread([this, store]() {
 
-        while (true) {
+        loopUntilInterrupted([this, store]() {
```

Comment:
It's not really clear to me why we don't just call `checkInterrupt()` in the while loop.

The issue of having an uncaught exception is orthogonal to the interruption issue. It's better to have an appropriate `try { ... } catch (...) { ... }` at top-level in the thread.

Reply:
Here's an alternate fix, https://github.com/NixOS/nix/pull/11665/commits/31fb9a47c5c26b03e9a719aceb31f5c9e83ac980

See also my accidental quote reply https://github.com/NixOS/nix/pull/11665#issuecomment-2404853301